### ISSUE TITLE
Problem with direct fm losses  when layers are asymmetric 

### ISSUE THREAD

**Issue by johcarter** [OasisLMF/ktools#33]
When the number of layers vary by final level agg_id in the policytc file, there are some missing output ids in the fm output stream. This skews the allocated losses of the output_ids that are present resulting in fm loss > gul.

A workaround for allocrule1 and 0 is to add in dummy layers for all final level agg_ids in the fm_policytc file so that each final level agg_id has the same number of layers. Use calcrule 14 with 0 limit for the dummy layers to ensure loss = 0.

This does not fix allocrule 2 losses which have a separate issue.
@benhayes21 FYI

**Comment by johcarter** [comment]
Use test case fm43 in ftest for testing fix @bmatharu

**Comment by johcarter** [comment]
tested and fixed.

**Comment by johcarter** [comment]
this problem seemed to have reemerged on a different data set in 3.0.7. however it was found to be a problem on executables built and run in ubuntu 1604 due to using an old gcc compiler (v5). if the executables are built and run in ubuntu 1804 (gcc v7.3) or in windows msys64 then there is no problem.
